app/components/session-offerings.js

```diff
@@ -3,28 +3,153 @@ import Ember from 'ember';
 import layout from '../templates/components/session-offerings';
 import { translationMacro as t } from "ember-i18n";
 
-export default Ember.Component.extend({
-  store: Ember.inject.service(),
-  i18n: Ember.inject.service(),
-  layout: layout,
+const { Component, computed, inject, isPresent, RSVP } = Ember;
+const { service } = inject;
+const { alias, oneWay } = computed;
+const { all } = RSVP;
+
+export default Component.extend({
+  saving: false,
+
+  store: service(),
+
+  i18n: service(),
+
+  layout,
+
   classNames: ['session-offerings'],
+
   session: null,
+
   placeholderValue: t('sessions.titleFilterPlaceholder'),
-  offerings: Ember.computed.oneWay('session.offerings'),
+
+  offerings: oneWay('session.offerings'),
+
   newButtonTitle: t('general.add'),
-  newOfferings: [],
+
+  offeringEditorOn: false,
+
+  cohorts: alias('session.course.cohorts'),
+
   actions: {
-    add: function(){
-      var self = this;
-      var offering = this.get('store').createRecord('offering', {
-        session: self.get('session'),
-        startDate: moment().hour(8).minute(0).second(0).format(),
-        endDate: moment().hour(9).minute(0).second(0).format(),
+    addSingleOffering({ startDate, endDate, room, learnerGroups, instructors, instructorGroups }) {
+      const store = this.get('store');
+      const session = this.get('session');
+      const offering = store.createRecord('offering', { session, startDate, endDate, room });
```

Comment:
this should be let instead of const.  The object isn't getting modified, but it isn't really a constant.  JS seems to be a little wishy washy right now about how this works, but I would prefer a let here.